docs/troubleshoot/app-framework/appserver-error-codes.md

```diff
@@ -1800,291 +1800,348 @@ The following error message codes may appear on the app-server log. Use the foll
 
   **Reason:**
 
-  TODO
+  Failed to add proxy authorization with reason _errorMessage_.
 
   **Action:**
 
-  TODO
+  No action required. If your desired authentication plugin isn't successfully authenticating a login,
+  please send the log and any relevant info to the developers.
+
 
 
+### ZWED0069W
 
-### ZWED0069W":"Returning null for cipher array because input had non-string: 
+  Returning null for cipher array because input had non-string:
 
   **Reason:**
 
-  TODO
+  Returns null for cipher array if an array element is not a string type.
 
   **Action:**
 
-  TODO
+  As it is an array of string then all elements should be of type string.
 
 
 
-### ZWED0070W":"Error when reading PFX. Server cannot continue. Error=%s
+### ZWED0070W
+
+  Error when reading PFX. The server cannot continue. Error=_errorMessage_
 
   **Reason:**
 
-  TODO
+  If we get an error while reading _config.https.pfx_ file then the server cannot continue
+  and throws _errorMessage_ in the catch block.
 
   **Action:**
 
-  TODO
+  No action is needed, but the _errorMessage_ may be needed to debug
 
 
 
-### ZWED0071W":"Unexpected error on server %s:%s. E=%s. Stack trace follows.\n%s
+### ZWED0071W
+
+  Unexpected error on server _ipAddress_: _port_ Error=_errorMessage_ stack trace Exception
 
   **Reason:**
 
-  TODO
+  When we get an unexpected error on the server with _ipAddress_ and _port_.
 
   **Action:**
 
-  TODO
+  No action needed unless user is experiencing an interruption in server, then send stack to developers
+
 
 
+### ZWED0072W
 
-### ZWED0072W":"Could not stop manager, error=
+  Could not stop manager due to error _errorMessage_
 
   **Reason:**
 
-  TODO
+  If the server manager is unable to stop due to any reason it will throw an exception with an _errorMessage_.
 
   **Action:**
 
-  TODO
+  Review the _errorMessage_ and see if there is something you can do to fix the error
 
 
 
-### ZWED0073W":"No server returned for group=
+### ZWED0073W
+
+  No server returned for group=_group_
 
   **Reason:**
 
-  TODO
+  If no server returned for _group_ element from _groupingConfig_ array
+  then it will throw this warning message.
 
   **Action:**
 
-  TODO
+  Review the warning message and check for the _group_ element to fix the error
 
 
+### ZWED0074W
 
-### ZWED0074W":"Unknown default behavior=%s
+  Unknown default behavior=_defaultBehavior_
 
   **Reason:**
 
-  TODO
+  If the server is not in a microservice or appserver then it logs a default
+  switch case warning message as the unknown _defaultBehavior_.
 
   **Action:**
 
-  TODO
+  No action is needed, but the warning may be needed to debug
 
 
 
-### ZWED0075W":"Services in plugin=%s war grouping skipped. Plugin missing or already grouped
+### ZWED0075W
+
+  Services in plugin=_plugin[i]_ war grouping skipped. Plugin missing or already grouped
 
   **Reason:**
 
-  TODO
+  If element from _plugins_ array, does not contain a compatible service (data service plugin/war service)
```

Comment:
_plugins_ isn't a variable here

Reply:
This is a little too technical I think. 
Reason: 
Server was not created for _plugin[i]_ War grouping, because it was already made or plugin is missing.